### ISSUE TITLE
Ark Server update does not work.

### ISSUE THREAD

**Issue by Fedli123** [GameServerManagers/LinuxGSM#3079]
Hello, can any help me with this error I have?

My arkserver runs with ark version  313.57 I want to update it to 314.5 with the `./arkserver update` command, but I get this error: 
```
[ FAIL ] Updating arkserver: Checking remote build: SteamCMD
arkserver@v2202007106712123820:~$
```
I have the LGSM version: v20.5.1 and never had any errors with updating the server, until now.

Here are the server details, because the server runs on version 313.57:
```
Distro Details
=================================
Distro:    Debian GNU/Linux 10 (buster)
Arch:      x86_64
Kernel:    4.19.0-9-amd64
Hostname:  v2202007106712123820
Uptime:    0d, 9h, 44m
tmux:      tmux 2.8
glibc:     2.28

Server Resource
=================================
CPU
Model:      AMD EPYC 7702P 64-Core Processor
Cores:      4
Frequency:  1996.238MHz
Avg Load:   0.63, 0.25, 0.12

Memory
Mem:       total  used   free  cached  available
Physical:  16GB   1.1GB  15GB  5.9GB   15GB
Swap:      0B     0B     0B

Storage
Filesystem:  /dev/sda3
Total:       157G
Used:        54G
Available:   97G

Network
Interface:  eth0
IP:         xx.xx.xx.xx

Game Server Resource Usage
=================================
CPU Used:  25.1%
Mem Used:  0.7%   128MB

Storage
Total:        23G
Serverfiles:  20G
Backups:      56K

ARK: Survival Evolved Server Details
=================================
Server name:      xxxxxxxxxxxxxx
App ID:           376030
Server IP:        xxxxxxxxxxxxxx
Server password:  NOT SET
Admin password:   xxxxxxxxxxxxxx
Maxplayers:       70
Default map:      TheIsland
Master server:    not listed
Status:           ONLINE

arkserver Script Details
=================================
Script name:            arkserver
LinuxGSM version:       v20.5.1
glibc required:         2.15
Discord alert:          off
Email alert:            off
IFTTT alert:            off
Mailgun (email) alert:  off
Pushbullet alert:       off
Pushover alert:         off
Rocketchat alert:       off
Slack alert:            off
Telegram alert:         off
Update on start:        off
User:                   arkserver
Location:               /home/arkserver
Config file:            /home/arkserver/serverfiles/ShooterGame/Saved/Config/LinuxServer/GameUserSettings.ini

Backups
=================================
No Backups created

Command-line Parameters
=================================
./ShooterGameServer TheIsland?AltSaveDirectoryName=TheIsland?listen?MultiHome=xx.xx.xx.xx?MaxPlayers=70?QueryPort=27016?RCONPort=27021?Port=7779?ServerAdminPassword=xxxxxxxxxxxxxx -automanagedmods -crossplay -PublicIPForEpic=xx.xx.xx.xx

Ports
=================================
Change ports by editing the parameters in:
/home/arkserver/lgsm/config-lgsm/arkserver

Useful port diagnostic command:
netstat -atunp | grep ShooterGame

DESCRIPTION  DIRECTION  PORT   PROTOCOL
> Game       INBOUND    7779   udp
> RAW        INBOUND    7780   udp
> Query      INBOUND    27016  udp
> RCON       INBOUND    27021  tcp

Status: ONLINE
```
Thanks for helping me :-)

**Comment by dgibbs64** [comment]
For whatever reason, steamcmd is not providing the version number. Try validate command instead this might force the update

**Comment by Fedli123** [comment]
Hello dgibbs64,

i try that and i delete all files and installed the server new, but i have the same error. 
i can update all servers i run on the rootserver (CS:S,DOD:S,ArmA3) without any problems, its only the arkserver that makes me that errors.

./arkserver update-lgsm
./arkserver validate
./arkserver force-update
./arkserver update

than i have the same error:

arkserver@v2202007106712123820:~$ ./arkserver update-lgsm
[  OK  ] Updating LinuxGSM arkserver: Selecting repo: GitHub
checking GitHub linuxgsm.sh...OK
checking arkserver...OK
checking GitHub config _default.cfg...OK
checking GitHub module check_deps.sh...OK
checking GitHub module check_glibc.sh...OK
checking GitHub module check_last_update.sh...OK
checking GitHub module check_logs.sh...OK
checking GitHub module check_permissions.sh...OK
checking GitHub module check_root.sh...OK
checking GitHub module check.sh...OK
checking GitHub module check_status.sh...OK
checking GitHub module check_steamcmd.sh...OK
checking GitHub module check_system_dir.sh...OK
checking GitHub module check_system_requirements.sh...OK
checking GitHub module check_tmuxception.sh...OK
checking GitHub module check_version.sh...OK
checking GitHub module command_install.sh...OK
checking GitHub module command_update_linuxgsm.sh...OK
checking GitHub module command_update.sh...OK
checking GitHub module command_validate.sh...OK
checking GitHub module core_dl.sh...OK
checking GitHub module core_exit.sh...OK
checking GitHub module core_functions.sh...OK
checking GitHub module core_getopt.sh...OK
checking GitHub module core_legacy.sh...OK
checking GitHub module core_logs.sh...OK
checking GitHub module core_messages.sh...OK
checking GitHub module core_trap.sh...OK
checking GitHub module fix.sh...OK
checking GitHub module info_distro.sh...OK
checking GitHub module install_complete.sh...OK
checking GitHub module install_config.sh...OK
checking GitHub module install_header.sh...OK
checking GitHub module install_logs.sh...OK
checking GitHub module install_server_dir.sh...OK
checking GitHub module install_server_files.sh...OK
checking GitHub module install_stats.sh...OK
checking GitHub module install_steamcmd.sh...OK
checking GitHub module update_steamcmd.sh...OK
[  OK  ] Updating LinuxGSM arkserver: Updating functions
arkserver@v2202007106712123820:~$ ./arkserver validate
[ WARN ] Validating arkserver: Validate might overwrite some customised files
[ START ] Validating arkserver: SteamCMD
Redirecting stderr to '/home/arkserver/.local/share/Steam/logs/stderr.txt'
Complete! Validating arkserver: SteamCMD
arkserver@v2202007106712123820:~$ ./arkserver force-update
[ START ] Updating arkserver: SteamCMD
Redirecting stderr to '/home/arkserver/.local/share/Steam/logs/stderr.txt'
Complete! Updating arkserver: SteamCMD
arkserver@v2202007106712123820:~$ ./arkserver update
[ FAIL ] Updating arkserver: Checking remote build: SteamCMDarkserver@v2202007106712123820:~$

**Comment by Fedli123** [comment]
Hello dgibbs64,

i found the error, its the new LGSM version, because i installed the older one (v20.4.1) and with this version i have no errors.

**Comment by h3o66** [comment]
For me it seems to be a problem wirth the steamcmd in general.
Could you provide the logs from `~/.local/share/Steam/logs/` and maybe you can as well provide a `dev-debug.log` of lgsm ?

To Produce a `dev-debug.log` you need to do the following:
```
# Enable dev debug
./arkserver dev
# execute the command, that does make problems:
./arkserver update
# disable dev debug again
./arkserver dev
```

Or if you want to do the debugging from another server like CSGO then just do the same above and replace the arkserver with the other gameserver management script name.

Maybe this helps us to pin down the problem a bit better.

**Comment by Fedli123** [comment]
Hello h3o66, 

we play at the moment every evening, but if the server is empty i will do that. :-)

**Comment by Fedli123** [comment]
Hello h3o66,

i can not test it, because the arkserver is up to date now, so the update progress is not starting. But i can test it if a new ark version is available.

**Comment by h3o66** [comment]
well, you could probably test it, if you switch the branch in config, then validate the files and remove the branch from the setting and update again.

You can get the branches from here again:
https://steamdb.info/app/376030/depots/

I only would propose to save your map data before doing this.

**Comment by SaintFenix** [comment]
I'm having the exact same issue as @Fedli123 did. I have attached the dev-debug logs, steam logs, and arkserver details for my server: 

Please let me know if you'd like me to place this as a separate issue.

-Fenix
[Steamlogs.zip](https://github.com/GameServerManagers/LinuxGSM/files/5706826/Steamlogs.zip)
[dev-debug-function-order.log](https://github.com/GameServerManagers/LinuxGSM/files/5706827/dev-debug-function-order.log)
[arkserverdetails.txt](https://github.com/GameServerManagers/LinuxGSM/files/5706829/arkserverdetails.txt)
[dev-debug.log](https://github.com/GameServerManagers/LinuxGSM/files/5706830/dev-debug.log)

**Comment by shinySonic** [comment]
i'm facing the same issue since yesterday:
`[ FAIL ] Updating arkserver: Checking remote build: SteamCMD`

**Comment by h3o66** [comment]
> i'm facing the same issue since yesterday:
> `[ FAIL ] Updating arkserver: Checking remote build: SteamCMD`

your problem is probably related to #3129

**Comment by SaintFenix** [comment]
@JoinedSenses figured out the fix and it worked for me:

> Fix for me is to change line 45 of home/ACCOUNT/lgsm/functions/update_steamcmd.sh to:
> `elif [ -n "${branch}" ]&&[ "${branch}" != "public" ]; then`

**Comment by thors10d** [comment]
> @JoinedSenses figured out the fix and it worked for me:
> 
> > Fix for me is to change line 45 of home/ACCOUNT/lgsm/functions/update_steamcmd.sh to:
> > `elif [ -n "${branch}" ]&&[ "${branch}" != "public" ]; then`

Yes, just to quickly confirm - this fixed it for me as well (updated from 319.22 to v320.18 just fine with it). Thanks for sharing!

**Comment by h3o66** [comment]
Closing this for now, please reopen if this this still persists
